src/app/outbound.rs

```diff
@@ -361,8 +361,13 @@ pub mod server_id {
                             warn!("l5d-server-id identity header is invalid: {:?}", endpoint);
                         }
                     },
-                    Conditional::None(_why) => {
-                        warn!("endpoint tls is some, but no identity: {:?}", endpoint);
+                    Conditional::None(why) => {
+                        error!("endpoint tls is some, but no identity ({:?}): {:?}", why, endpoint);
+
+                        // This is a bug, so panic in tests!
+                        if cfg!(debug_assertions) {
```

Comment:
Nice --- I wonder if there are other places where we might want to do this behavior. Maybe eventually we ought to add a `bug!` macro or something.

Reply:
There is/have been places where we've done `debug_assert!(false, "here's a bug you hit")`